### PR TITLE
FIX: HiDPI testing

### LINKED ISSUE #551: Visuals should make use of document coordinate system
Most visuals make the assumption that the document and framebuffer coordinate systems are the same. This can be seen most easily by running `examples/basics/plot/export.pt`  (requires #352) -- the exported image should look roughly the same as the original canvas, but currently the line width and marker sizes are fixed to the framebuffer pixels, rather than to document pixels. 

We can write simple unit tests for this by rendering a scene once at native resolution, and again at 2x resolution, and then checking `image1 == image2[::2, ::2]`. (this requires antialiasing to be disabled to get a good match).

### LINKED ISSUE #789: All visuals need to use transform or scaling factor to support HiDPI
You can't just plot a fixed number of pixels to screen when writing your own visuals: everything has to be scaled up or down. i.e. lines, which should never be left as 1px, and instead should be scaled with the transform.

Top image is glfw which is hi-dpi aware (in this case, `canvas.physical_size == canvas.size * 2`)
Bottom is qt4 which is not hi-dpi aware and hides this from the program (`canvas.physical_size == canvas.size`)

Both on Mac OS X 10.10 on a 15" Retina MBP.

screenshot 2015-03-26 10 46 49

### PR DESCRIPTION
**Pull request by Eric89GXL**
Closes #789.
Closes #551.

@nippoo can you see if this makes tests pass for you?

### CODE REVIEW

**Comment by Eric89GXL** [comment]
@bollu where you at? This should make your life less painful!

**Comment by Eric89GXL** [comment]
sorry @bollu, I meant to ping @nippoo with that one... @nippoo this should make your life less painful.

Although @bollu if you happen to be using a HiDPI screen it might also make your life less painful, too :)

**Comment by nippoo** [comment]
@Eric89GXL all tests pass :)

**Comment by Eric89GXL** [comment]
Excellent, @almarklein @rossant feel free to merge then

File: vispy/geometry/calculations.py [hunk 1]
```diff
@@ -78,3 +80,57 @@ def _calculate_normals(rr, tris):
...
+    shape : tuple
+        2-element shape.
+    kind : str
+        Interpolation, either "linear" or "nearest".
+
+    Returns
+    -------
+    scaled_image : ndarray
+        New image, will have dtype np.float64.
+    """
```
Comment (almarklein) [inline vispy/geometry/calculations.py]: Is there a reason to add this? I see it can be useful, but it does not seem like we're using it?

**Comment by almarklein** [comment]
Other than that, LGTM

File: vispy/geometry/calculations.py (hunk 1 above)
Comment (Eric89GXL) [inline vispy/geometry/calculations.py]: We are, it's used to rescale images to the testing image size

Comment (Eric89GXL) [inline vispy/geometry/calculations.py]: Right now we only use the `nearest` method, but the `linear` could come in handy in the future (I've wanted it before)

**Pull request merged**